assets/js/customizer.js

```diff
@@ -7,108 +7,124 @@
 				addItemButton = container.querySelector(".add-featured-item");
 			const { __, _x, _n, _nx } = wp.i18n;
 			let featuredArea,
-				itemObjects = new Map(),
-				timer,
+				settings_timer,
 				search_timer,
 				timer_ms = 500;
 
 			class ListItem {
-				constructor(post) {
-					this.key = post.ID;
+				constructor(post, parent) {
+					this.id = post.id;
 					this.postData = post;
-					this.featured_area = control.id;
-					this.element_id = "item_" + post.ID;
+					this.parent = parent;
+					this.element = null;
+					this.element_id = control.id + "_item_" + post.id;
 
-					// Add item element to list
+					// Add item element to ol list.
 					this.addItem();
 				}
 
-				// Create featured item from the Set and the DOM
+				// Create featured item element.
 				addItem() {
-					let item = document.getElementById(this.element_id),
-						featuredTtemTemplate = wp.template("featured-item");
-					if (!item) {
-						item = document.createElement("li");
-						item.classList.add(this.postData.original_post_status);
-						item.id = "item_" + this.key;
-						item.innerHTML = featuredTtemTemplate(this.postData); // WP templating the markup
-						item
+					let featuredItemTemplate = wp.template("featured-item");
+					this.element = areaContainer.querySelector('[data-id="' + this.id + '"]');
+					if (!this.element) {
+						this.element = document.createElement("li");
+						this.element.id = this.element_id;
+						this.element.classList.add(this.postData.post_status);
+						for(var attribute in this.postData ) {
+							this.element.setAttribute('data-' + attribute, this.postData[attribute]);
```

Comment:
Another small back-compat here would be to skip post_content if it exists, it's often very large and make the dom superslow when added as a data-attribute.

`if ( 'post_content' !== attribute ) {
								this.element.setAttribute('data-' + attribute, this.postData[attribute]);
							}`